katello_certs_tools/rhn_rpm.py

```diff
@@ -73,7 +75,7 @@ def __nonzero__(self):
 
     def checksum_type(self):
         if self.hdr[rpm.RPMTAG_FILEDIGESTALGO] \
-           and PGPHASHALGO.has_key(self.hdr[rpm.RPMTAG_FILEDIGESTALGO]):
+           and self.hdr[rpm.RPMTAG_FILEDIGESTALGO] in PGPHASHALGO:
```

Comment:
this probably also suffers from https://bugzilla.redhat.com/show_bug.cgi?id=1631292#c2 and needs some `decode` dance